packages/api/src/command/webhook/fitbit.ts

```diff
@@ -0,0 +1,176 @@
+import { EventTypes, analytics } from "../../shared/analytics";
+import { Constants } from "../../shared/constants";
+import { capture } from "../../shared/notifications";
+import { Util } from "../../shared/util";
+import { getConnectedUserByTokenOrFail } from "../connected-user/get-connected-user";
+import { getSettingsOrFail } from "../settings/getSettings";
+import { ApiTypes } from "../usage/report-usage";
+import {
+  DataType,
+  WebhookDataPayloadWithoutMessageId,
+  WebhookUserDataPayload,
+  processRequest,
+  reportDevicesUsage,
+} from "./webhook";
+import { createWebhookRequest } from "./webhook-request";
+
+export type FitbitWebhookNotification = {
+  collectionType: FitbitCollectionTypes;
+  date: string;
+  ownerId: string;
+  ownerType: string;
+  subscriptionId: string;
+};
+
+interface Entry {
+  cxId: string;
+  userId: string;
+  typedData: WebhookUserDataPayload;
+}
+
+const log = Util.log(`Fitbit Webhook`);
+
+export const processData = async (data: FitbitWebhookNotification[]): Promise<void> => {
+  console.log("Starting to process the webhook");
+
+  const connectedUsersAndData = await Promise.all(
+    data.map(async d => {
+      const { collectionType, date, ownerId: fitbitUserId } = d;
+      const connectedUser = await getConnectedUserByTokenOrFail(
+        ProviderSource.fitbit,
+        fitbitUserId
+      );
+
+      const userFitbitData = await mapData(collectionType, connectedUser, date);
+      return { cxId: connectedUser.cxId, userId: connectedUser.id, typedData: userFitbitData };
+    })
+  );
+
+  const reducedData: Entry[] = [];
+
+  connectedUsersAndData.forEach(entry => {
+    const existingUserIndex = reducedData.findIndex(
+      item => item.cxId === entry.cxId && item.userId === entry.userId
+    );
+
+    if (existingUserIndex >= 0) {
+      const entryKeys = Object.keys(entry.typedData) as DataType[];
+
+      reducedData[existingUserIndex] = {
+        ...reducedData[existingUserIndex],
+        typedData: {
+          ...reducedData[existingUserIndex].typedData,
+          ...(entryKeys.includes("activity")
+            ? {
+                activity: union(
+                  reducedData[existingUserIndex].typedData.activity,
+                  entry.typedData.activity
+                ),
+              }
+            : undefined),
+          ...(entryKeys.includes("nutrition")
+            ? {
+                nutrition: union(
+                  reducedData[existingUserIndex].typedData.nutrition,
+                  entry.typedData.nutrition
+                ),
+              }
+            : undefined),
+          ...(entryKeys.includes("body")
+            ? { body: union(reducedData[existingUserIndex].typedData.body, entry.typedData.body) }
+            : undefined),
+          ...(entryKeys.includes("sleep")
+            ? {
+                sleep: union(reducedData[existingUserIndex].typedData.sleep, entry.typedData.sleep),
+              }
+            : undefined),
+        },
```

Comment:
NABD, but I was looking at this and wondering if we could do something to prevent possible runtime errors with this logic. Its sound, but it's also very easy to miss in case we add another WH data type (along `nutrition`, `body`, `sleep`).

We could have a specific type for Fitbit, which would be a subset of the potential WH types, and/or we could just unit test this function.

Again, NABD if you're rushing to get this out of the door, but the unit test would be good and this is something I've been missing on the PRs recently - we did decide to add those as we go, so in theory every PR should add unit tests for the main pieces it adds/fixes. 🤓